stubgen_wrapper.py

```diff
@@ -108,13 +110,29 @@ def wrapper():
         idx = args.index("-o")
         args[idx + 1] = str(bindir / args[idx + 1])
 
+    if "-O" in args:
+        # we have an output directory, use its path instead relative to $(BINDIR),
+        # but in absolute form.
+        idx = args.index("-O")
+        output_dir = args[idx + 1]
+        args[idx + 1] = str(bindir / args[idx + 1])
+    else:
+        output_dir = None
+
     if "-M" in args:
         # fix up the path to the marker file relative to $(BINDIR).
         idx = args.index("-M")
         args[idx + 1] = str(bindir / args[idx + 1])
 
     main(args)
 
+    if "-O" in args:
+        # TODO: Not sure what is the best way to handle this, using the Nanobind Bazel example,
+        # the root stub looks like _main.src.nanobind_example.pyi...
+        shutil.move(Path(bindir) / f'{output_dir}/_main.{output_dir.replace("/", ".")}.pyi', Path(bindir) / f'{output_dir}/__init__.pyi')
```

Comment:
I remember encountering this when attempting to write stubs relative to the working directory in a Bazel project - I think you cannot avoid botched stub names when specifying the output directory and it's not the same as the current working directory.

What I would like is this:

```python
src/nanobind_example/__init__.py
src/nanobind_example/nanobind_example_ext.pyi
src/nanobind_example/sub_ext/__init__.py
src/nanobind_example/sub_ext/__init__.pyi # <- this would contain stubs for sub()
```

I'll still have to check what stubs are actually produced with your changes, but tl,dr: I think the correct solution is that top-level module extensions get named stubs, and submodules get `__init__.pyi` files.

Maybe we can avoid the move hack by tweaking some logic in nanobind.stubgen, I'll check it out.